src/util/evalUnits.ts

```diff
@@ -53,19 +81,26 @@ export enum EvalMessages {
  * the first is the actual calculated number (or null if none) while
  * the second is the message that should appear in case of error
  */
-export function evalUnits(input: string): [number | null, string] {
+export function evalUnits(input: string): [BN | number | null, string] {
   if (!floats.test(input) && !ints.test(input) && !alphaInts.test(input) && !alphaFloats.test(input)) {
     return [null, EvalMessages.GIBBERISH];
   }
-
   if (floats.test(input) || ints.test(input)) {
-    return [parseFloat(input.replace(/[,]/g, '.')), EvalMessages.SUCCESS];
-  } else if (alphaInts.test(input) || alphaFloats.test(input)) {
-    const numericPart = parseFloat(input.replace(/[,]/g, '.'));
-    const charPart = input.replace(/[0-9.,]/g, '');
-    const siVal = si.find((s) => s.symbol === charPart);
-    return siVal ? [numericPart * siVal.value, EvalMessages.SUCCESS] : [null, EvalMessages.SYMBOL_ERROR];
-  } else {
+    const result = parseFloat(input.replace(/[,]/g, '.'));
+    return [result, EvalMessages.SUCCESS];
+  }
+  if (alphaInts.test(input) || alphaFloats.test(input)) {
```

Comment:
Can you give some examples that don't pass this test and requires the fallback below?

Reply:
Code should never reach there. It was either a `"never reachable" return` or add to return types `undefined` in order to satisfy TSlint.
I will alter it and leave last option the ZERO one.